### PR TITLE
Patch for terraform AWS provider issue #6715

### PR DESCRIPTION
**Pull request by userhas404d**
Workaround for [AWS Provider version #6715](https://github.com/terraform-providers/terraform-provider-aws/issues/6715)

### CODE REVIEW

**Comment by userhas404d** [comment]
Note: pinning to provider version `1.43.0` and lower will prevent this error without requiring changes to the existing module.

Our org has given the aws provider maintainers a [heads up](https://github.com/terraform-providers/terraform-provider-aws/pull/5762#issuecomment-447483707) regarding this issue in the current aws provider so hopefully we see a fix implemented soon.

**Comment by thrinadh238** [comment]
Hi 

I need some information regarding this issue.

The above change is because of upgrading Terraform version  or AWS made it mandatory???

**Comment by userhas404d** [comment]
@thrinadh238 a change was introduced in the terraform aws provider (v1.44.0 - see the 'heads up' link in my other comment for more details) that added the `name` and `description` options to the `aws_ssm_maintenance_window_task` resource. A welcome change, but one that does not allow them to be added conditionally (both are optional [according to AWS](https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_RegisterTaskWithMaintenanceWindow.html#API_RegisterTaskWithMaintenanceWindow_RequestSyntax)). Pinning the terraform aws provider to versions less than 1.43 will avoid this issue unless they are explicitly defined in the resource definition as laid out in this PR. 

TL;DR: This change is required because of a bug introduced in v1.44.0 of the terraform aws provider

**Comment by thrinadh238** [comment]
Hi,

Thanks for the quick reply.

Terraform documentation shows that it is optional.  Please find the image below attached.

The version of AWS provider which you mentioned above is 1.43 but, we tried with below version 
version               = "~> 1.48.0".

It actually worked. It is not working with version 1.50

capture
